docs/DeveloperGuide.md

```diff
@@ -114,7 +114,27 @@ The features of WerkIt! are split and grouped into 5 main features:
 5. [Search-related features](#search-related-features)
 
 ### Exercise-related features
-_to be updated_
+
+Format: `exercise <userAction> <keywords>`
+
+Below is a class diagram of the exercise-related features:
+
+![ExerciseUML](uml/classDiagrams/images/exercise.png)
+<br>
+
+When WerkIt is running, the `WerkIt` class will keep prompting the user to enter command through the
+`WerkIt#startContinuousUserPrompt()` method. After the user has entered command, The `UI#getUserInput()` method in `UI`
+class will catch the user input, and it will be sent to `Parser#parseUserInput(String userInput)` method to analyse the
+user's command. If the user's command type is `exercise`, the `Parser#parseUserInput(String userInput)` method will 
+parse the 'exercise' base word and proceed to create exercise related command using 
+`Parser#createExerciseCommand(String userInput)` method. This method will further evaluate the
+`<userAction>` and call the constructor of `SearchCommand` class by passing relevant parameters related to search to the
+constructor. If the `<userAction>` is null or incorrect, an `InvalidCommandException` will be thrown.
```

Comment:
Line 131: is `SearchCommand` supposed to be there?

Reply:
Oh, my mistake, will fix it soon.